go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,304 @@
+		oc.deleteEgressFirewall(oldEgressFirewall)
+		oc.addEgressFirewall(newEgressFirewall)
+	}
+}
+
+func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall) {
+	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
+
+	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
+	if err != nil {
+		klog.Errorf("Failed to wait for namespace %s event (%v)",
+			egressFirewall.Namespace, err)
+	}
+	defer nsInfo.Unlock()
+	nsInfo.egressFirewall = false
+
+	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid", "find", "ACL",
+		fmt.Sprintf("external-ids:egressFirewall=%s", egressFirewall.Namespace))
+	if err != nil {
+		klog.Errorf(" error executing create ACL command, stderr: %q, %+v", stderr, err)
+		//none of the elements of the egressfirewall have been deleted
+		nsInfo.egressFirewall = true
+		return
+	}
+	uuids := strings.Fields(stdout)
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		klog.Errorf("Error deleting egressFirewall for namespace %s, cannot get nodes to delete ACLS %v",
+			egressFirewall.Namespace, err)
+	}
+	for _, uuid := range uuids {
+		for _, node := range existingNodes.Items {
+			_, stderr, err := util.RunOVNNbctl("remove", "logical_switch",
+				joinSwitch(node.Name), "acls", uuid)
+			if err != nil {
+				klog.Errorf("Remove failed to delete the rule for "+
+					"address_set=%s, stderr: %q (%v)", nsInfo.addressSet.GetIPv4HashName(), stderr, err)
+			}
+		}
+	}
+}
+
+func (ef *egressFirewall) addACLToJoinSwitch(joinSwitch, hashedAddressSetName string) error {
+	for _, rule := range ef.egressRules {
+		var match string
+		var action string
+		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
+			action = "allow"
+		} else {
+			action = "drop"
+		}
+
+		if len(rule.ports) == 0 {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if ipAddress.To4() != nil {
+				//rule.to.cidrSelector is an ipv4 address
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetName)
+			} else {
+				//rule.to.cidrSelector is an ipv6 address
+				match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetName)
+			}
+		} else {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if ipAddress.To4() != nil {
+				//rule.to.cidrSelector is an ipv4 address
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s && ( %s )\"", rule.to.cidrSelector, hashedAddressSetName, egressGetL4Match(rule.ports))
+			} else {
+				//rule.to.cidrSelector is an ipv6 address
```

Comment:
All these comments are quite useless. The code is quite evident here.

Reply:
done